### PR TITLE
Add test for remaining axis options

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

These three were not tested. ~Selected data so that they would show up slightly different at least.~

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

**Comment by QuLogic** [comment]
Since these options talk about making circles circular, I wonder if it might be clearer to plot circles here?

**Comment by oscargus** [comment]
I've added a circle as well. Cannot really find a case where `'scaled'` and `'image'` differs though.

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -7280,6 +7280,20 @@ def test_secondary_repr():
     assert repr(secax) == '<SecondaryAxis: >'
 
 
+@image_comparison(['axis_options.png'], remove_text=True, style='mpl20')
+def test_axis_options():
+    fig, axes = plt.subplots(2, 3)
+    for i, option in enumerate(('scaled', 'tight', 'image')):
+        axes[0, i].plot((1, 2), (1, 3.2))
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_axes.py:7287]: What's thd difference between the top and bottom tests?

Comment (oscargus) [reply lib/matplotlib/tests/test_axes.py:7287]: Aspect ratio.

Comment (timhoffm) [reply lib/matplotlib/tests/test_axes.py:7287]: Is that a relevant parameter to test? I would have expected that the code paths are the same.

If it is, please add comments á la `# aspect ratio > 1` to both parts.

Comment (tacaswell) [reply lib/matplotlib/tests/test_axes.py:7287]: Is the aspect ratio coming from just the autoscaling on the line?

Testing both of these seems relevant, the Do What I Mean code tends to be full of (and grow) special cases so (low fidelity) fuzz testing is reasonable.

**Review by tacaswell — APPROVED** [review]
I have some confusion about how the aspect ratio is being controlled.  The confusion is enough I do not want to push the green button, but little enough I am :+1: on someone who does not feel confused pushing the button.

**Review by dstansby — COMMENTED** [review]
Could you stick some comments in the test about why the different values being plotted are chosen? Appreciate this is testing new code so is valuable as is (and I won't blcok), but the test without any comments is a bit confusing to me.

File: lib/matplotlib/tests/test_axes.py (hunk 1 above)
Comment (oscargus) [reply lib/matplotlib/tests/test_axes.py:7287]: Yes, that is my impression. It will at least not consider a large circle.

**Comment by oscargus** [comment]
I've added a bit of comments.

The idea was basically to just try it with a few different scenarios to make sure that we do not break it later. At least not unless it is on purpose.

(Also, I like high coverage numbers...)

**Review by dstansby — APPROVED** [review]

**Pull request merged**